src/Symfony/Component/Cache/Adapter/CouchbaseBucketAdapter.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Couchbase\Bucket;
+use Symfony\Component\Cache\Exception\CacheException;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+use Symfony\Component\Cache\Marshaller\DefaultMarshaller;
+use Symfony\Component\Cache\Marshaller\MarshallerInterface;
+
+/**
+ * Class CouchbaseBucketAdapter.
+ */
+class CouchbaseBucketAdapter extends AbstractAdapter
+{
+    private const THIRTY_DAYS_IN_SECONDS = 2592000;
+    private const MAX_KEY_LENGTH = 250;
+    private const KEY_NOT_FOUND = 13;
+
+    /** @var Bucket */
+    private $bucket;
+
+    /** @var MarshallerInterface */
+    private $marshaller;
+
+    public function __construct(\CouchbaseCluster $client, string $bucket, string $namespace = '', int $defaultLifetime = 0, array $options = [])
+    {
+        $this->init($client, $bucket, $namespace, $defaultLifetime, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function createConnection($servers, array $options = []): \CouchbaseCluster
```

Comment:
this should return a bucket if you agree with my previous comment

Reply:
Ok.